File: JavaProjects/TemplateMethodSolucao/src/metodoPagamento/PagamentoCredito.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package metodoPagamento;

import gatewayPagamento.Gateway;

/**
 *
 * @author jeffe
 */
public class PagamentoCredito extends Pagamento {

    public PagamentoCredito(double valor, Gateway gateway) {
        super.valor = valor;
        super.gateway = gateway;
    }
    
    @Override
    public double calcularTaxa(){
        return super.valor*0.05;
    }
    
    @Override
    public double calcularDesconto(){
        if(super.valor>300)
            return super.valor*0.02;
        else return 0;
    }
}

File: CsharpProjects/TemplateMethodSolucao/TemplateMethodSolucao/MetodoPagamento/PagamentoCredito.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TemplateMethodSolucao.GatewayCobranca;

namespace TemplateMethodSolucao.MetodoPagamento
{
    public class PagamentoCredito : Pagamento
    {
        public PagamentoCredito(double valor, Gateway gateway)
        {
            base.valor = valor;
            base.gateway = gateway;
        }

        public override double calcularTaxa()
        {
            return this.valor * 0.05;
        }

        public override double calcularDesconto()
        {
            if (this.valor > 300)
                return this.valor * 0.02;
            else return 0;
        }
    }
}
